lib/state/action/pokemon_actions.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:async_redux/async_redux.dart';
+import 'package:pokedex/model/dto/pokemon_dto.dart';
+import 'package:pokedex/state/app_state.dart';
+import 'package:pokedex/api/api_service.dart';
+import 'package:pokedex/utils/constant.dart' as k;
+import 'package:pokedex/utils/extensions.dart';
+
+class GetPokemonListAction extends ReduxAction<AppState> {
+  @override
+  Future<AppState> reduce() async {
+    final pokemonResponse = await ApiService().pokemonApi.getPokemonList(k.pokemonOffset, k.pokemonLimit);
+    final pokemonDto = pokemonResponse.map((pokemon) => PokemonDto(pokemon: pokemon, id: pokemon.url.getPokemonId)).toList();
+    return state.copyWith(pokemons: pokemonDto);
+  }
+}
+
+class GetPokemonAction extends ReduxAction<AppState> {
```

Comment:
also a reminder ⚠️ , always provide a **`DOCUMENTATION`** at the top of every action classes like what I've told you before => it will serve as a reference to look at for devs to see what a specific action in opt to implement 🙂